buildSrc/src/main/java/quilt/internal/tasks/setup/ExtractServerJarTask.java

```diff
@@ -1,35 +1,37 @@
 package quilt.internal.tasks.setup;
 
-import java.io.File;
 import java.io.IOException;
 
 import org.apache.commons.io.FileUtils;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFile;
 import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
 import quilt.internal.Constants;
 import quilt.internal.tasks.DefaultMappingsTask;
 
-public class ExtractServerJarTask extends DefaultMappingsTask {
+public abstract class ExtractServerJarTask extends DefaultMappingsTask {
     public static final String TASK_NAME = "extractServerJar";
 
+    @InputFile
+    public abstract RegularFileProperty getServerBootstrapJar();
+
     @OutputFile
-    private final File serverJar;
+    public abstract RegularFileProperty getServerJar();
 
     public ExtractServerJarTask() {
         super(Constants.Groups.SETUP_GROUP);
-        dependsOn(DownloadMinecraftJarsTask.TASK_NAME);
-
-        getInputs().file(this.getTaskByType(DownloadMinecraftJarsTask.class).getServerBootstrapJar());
-
-        serverJar = new File(fileConstants.cacheFilesMinecraft, Constants.MINECRAFT_VERSION + "-server.jar");
     }
 
     @TaskAction
     public void extractServerJar() throws IOException {
-        FileUtils.copyFile(getProject().zipTree(this.getTaskByType(DownloadMinecraftJarsTask.class).getServerBootstrapJar()).matching(patternFilterable -> patternFilterable.include("META-INF/versions/*/server-*.jar")).getSingleFile(), serverJar);
-    }
-
-    public File getServerJar() {
-        return serverJar;
+        FileUtils.copyFile(
+            // TODO eliminate project access in task action
+            this.getProject()
+                .zipTree(this.getServerBootstrapJar().get())
```

Comment:
Ah yes, our good friend `zipTree` that for some reason is not available on any non-internal injectable service -- this sucks unfortunately, and there's no nice solution other than building your own zip utility or splitting the task. Or possibly doing it in the task ctor? No clue if that laziness is lazy enough to work fine.

Reply:
Yeah I've updated another task to save the `zipTree` in the constructor and it's working fine

Reply:
The issue I asked about on discord was because the other task was also *outputting* the file that was getting unzipped, so I had to separate the unzipping to another task.